lib/ansible/parsing/yaml/objects.py

```diff
@@ -55,7 +55,18 @@ def _set_ansible_position(self, obj):
 
 class AnsibleMapping(AnsibleBaseYAMLObject, dict):
     ''' sub class for dictionaries '''
-    pass
+    mergereplace = None
+
+    def __init__(self, *args, **kwargs):
+        ''' This constructor is here mostly for testing. '''
+        self.mergereplace = kwargs.pop('__mergereplace__', None)
```

Comment:
I think we are fine with having the property and can do w/o the 'special dict key' which would show in the user's dict definition